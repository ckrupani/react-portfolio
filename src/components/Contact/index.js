import React, { useState } from "react";
import { validateEmail } from "../../utils/helpers";

function ContactForm({ title }) {
    const [formState, setFormState] = useState({
        name: "",
        email: "",
        message: "",
    });
    const [errorMessage, setErrorMessage] = useState("");
    const { name, email, message } = formState;

    function handleChange(e) {
        if (e.target.name === "email") {
            const isValid = validateEmail(e.target.value);
            console.log(isValid);
            if (!isValid) {
                setErrorMessage("Your email is invalid.");
            } else {
                setErrorMessage("");
            }
        } else {
            if (!e.target.value.length) {
                setErrorMessage(`${e.target.name} is required.`);
            } else {
                setErrorMessage("");
            }
        }

        if (!errorMessage) {
            setFormState({ ...formState, [e.target.name]: e.target.value });
        }
    }

    function handleSubmit(e) {
        e.preventDefault();
        console.log(formState);
    }

    return (
        <div>
            <h1 data-testid="contactForm">{title}</h1>
            <form id="contact-form" onSubmit={handleSubmit}>
                <div className="flex-row flex-vertical my-2">
                    <div>
                        <label htmlFor="name">Name:</label>
                    </div>
                    <div className="my-1">
                        <input
                            type="text"
                            name="name"
                            defaultValue={name}
                            onBlur={handleChange}
                        />
                    </div>
                </div>
                <div className="flex-row flex-vertical my-2">
                    <div>
                        <label htmlFor="email">Email address:</label>
                    </div>
                    <div>
                        <input
                            type="email"
                            name="email"
                            defaultValue={email}
                            onBlur={handleChange}
                        />
                    </div>
                </div>
                <div className="flex-row flex-vertical my-2">
                    <div>
                        <label htmlFor="message">Message:</label>
                    </div>
                    <div>
                        <textarea
                            name="message"
                            rows="5"
                            defaultValue={message}
                            onBlur={handleChange}
                        />
                    </div>
                </div>
                {errorMessage && (
                    <div>
                        <p className="error-text">{errorMessage}</p>
                    </div>
                )}
                <button data-testid="submitButton" type="submit">
                    Submit
                </button>
            </form>
        </div>
    );
}

export default ContactForm;
