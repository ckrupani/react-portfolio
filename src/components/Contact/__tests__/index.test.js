import React from "react";
import { render, cleanup, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Contact from "../";

afterEach(cleanup);

describe("Contact is rendering", () => {
    it("renders", () => {
        render(<Contact title={"Contact"} />);
    });

    it("matches the snapshot", () => {
        const { asFragment } = render(<Contact title={"Contact"} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("has correct header", () => {
        render(<Contact title={"Contact"} />);
        expect(screen.getByTestId("contactForm")).toHaveTextContent("Contact");
    });

    it("has label for submit button", () => {
        render(<Contact title={"Contact"} />);
        expect(screen.getByTestId("submitButton")).toHaveTextContent("Submit");
    });
});
