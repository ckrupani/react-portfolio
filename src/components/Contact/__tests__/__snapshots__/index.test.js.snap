// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact is rendering matches the snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1
      data-testid="contactForm"
    >
      Contact
    </h1>
    <form
      id="contact-form"
    >
      <div
        class="flex-row flex-vertical my-2"
      >
        <div>
          <label
            for="name"
          >
            Name:
          </label>
        </div>
        <div
          class="my-1"
        >
          <input
            name="name"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="flex-row flex-vertical my-2"
      >
        <div>
          <label
            for="email"
          >
            Email address:
          </label>
        </div>
        <div>
          <input
            name="email"
            type="email"
            value=""
          />
        </div>
      </div>
      <div
        class="flex-row flex-vertical my-2"
      >
        <div>
          <label
            for="message"
          >
            Message:
          </label>
        </div>
        <div>
          <textarea
            name="message"
            rows="5"
          />
        </div>
      </div>
      <button
        data-testid="submitButton"
        type="submit"
      >
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
